File: src/main/java/ru/greenbudgie/artifact/ArtifactDamage.java
package ru.greenbudgie.artifact;

import org.bukkit.Material;
import org.bukkit.Particle;
import org.bukkit.Sound;
import org.bukkit.entity.Player;
import ru.greenbudgie.UHC.PlayerManager;
import ru.greenbudgie.UHC.UHCPlayer;
import ru.greenbudgie.util.MathUtils;
import ru.greenbudgie.util.ParticleUtils;

import javax.annotation.Nullable;

public class ArtifactDamage extends Artifact {

    private static final int SELF_DAMAGE = 6;
    private static final int DAMAGE = 4;

    @Override
    public String getName() {
        return "Злодеяние";
    }

    @Override
    public String getDescription() {
        return "Наносит всем игрокам урон в 2 сердца. Использовавший игрок получает урон в 3 сердца. Умереть от этого артефакта нельзя.";
    }

    @Override
    public int getStartingPrice() {
        return 16;
    }

    @Override
    public float getPriceIncreaseAmount() {
        return 1.5F;
    }

    @Override
    public boolean onUse(@Nullable Player player) {
        for(UHCPlayer uhcCurrentPlayer : PlayerManager.getAlivePlayers()) {
            if(uhcCurrentPlayer.isOnline()) {
                Player currentPlayer = uhcCurrentPlayer.getPlayer();
                boolean isSelfDamage = player == currentPlayer;
                double damage = MathUtils.clamp(isSelfDamage ? SELF_DAMAGE : DAMAGE, 0, currentPlayer.getHealth() - 1);
                currentPlayer.damage(damage);
                ParticleUtils.createParticlesInRange(currentPlayer.getLocation(), 3, Particle.SMOKE_LARGE, null, 15);
            } else {
                double damage = MathUtils.clamp(DAMAGE, 0, uhcCurrentPlayer.getOfflineHealth() - 1);
                uhcCurrentPlayer.addOfflineHealth(-damage);
            }
        }
        for(Player currentPlayer : PlayerManager.getInGamePlayersAndSpectators()) {
            currentPlayer.playSound(currentPlayer.getLocation(), Sound.ENTITY_VILLAGER_NO, 1F, 0.5F);
        }
        return true;
    }

    @Override
    public Material getType() {
        return Material.FERMENTED_SPIDER_EYE;
    }

}

File: src/main/java/ru/greenbudgie/artifact/ArtifactDamage.kt
package ru.greenbudgie.artifact

import org.bukkit.Material
import org.bukkit.Particle
import org.bukkit.Sound
import org.bukkit.entity.Player
import ru.greenbudgie.UHC.PlayerManager
import ru.greenbudgie.util.DARK_GRAY
import ru.greenbudgie.util.MathUtils
import ru.greenbudgie.util.ParticleUtils

class ArtifactDamage : Artifact() {

    override fun getName() = "${DARK_GRAY}Злодеяние"

    override fun getDescription(): String {
        return "Дамажит всех игроков на 1.5 сердца. Использовавшего игрока дамажит на 2 сердца. Умереть нельзя."
    }

    override fun getStartingPrice() = 16

    override fun getPriceIncreaseAmount() = 1F

    override fun onUse(player: Player?): Boolean {
        PlayerManager.getAlivePlayers().forEach {
            if (it.isOnline) {
                val currentPlayer = it.player
                val doMaxDamage = player == null || player === currentPlayer
                val initialDamage = if (doMaxDamage) 4 else 3
                val damage = MathUtils.clamp(initialDamage.toDouble(), 0.0, currentPlayer.health - 1)
                currentPlayer.damage(damage)
                currentPlayer.playSound(currentPlayer.location, Sound.ENTITY_VILLAGER_NO, 1f, 0.5f)
                ParticleUtils.createParticlesInRange(currentPlayer.location, 3.0, Particle.SMOKE_LARGE, null, 15)
            } else {
                val damage = MathUtils.clamp(3.0, 0.0, it.offlineHealth - 1)
                it.addOfflineHealth(-damage)
            }
        }
        return true
    }

    override fun getType() = Material.FERMENTED_SPIDER_EYE
}
